### PR TITLE
Make compatible with pip 10

### LINKED ISSUE #203: pip 10 broken again: pip moved internal modules to `pip._internal`

### PR DESCRIPTION
**Pull request by chriskuehl**
Fixes #203

Mostly just updating the imports and changing some assertions about stdout/stderr to match newer pip's error messages.

Everything is passing in Travis now including latest-pip.

### CODE REVIEW

File: pip_faster.py [hunk 1]
```diff
@@ -383,12 +383,8 @@ def run(self, options, args):
...
             )
 
-        if requirement_set is None:
-            required = ()
-            successfully_installed = ()
-        else:
-            required = requirement_set.requirements.values()
-            successfully_installed = requirement_set.successfully_installed
+        required = requirement_set.requirements.values()
+        successfully_installed = requirement_set.successfully_downloaded
```
Comment (chriskuehl) [inline pip_faster.py]: I'm not 100% confident in this change, need to look into it more.

Two things going on:

* `successfully_installed` isn't on `RequirementSet` anymore, and the list of installed things doesn't seem to be made available at all (not returned in any way or attached to an object we have): https://github.com/pypa/pip/blob/60208552084ec16bf70b1d2913aed61cccdbe28d/src/pip/_internal/commands/install.py#L327-L337; successfully_downloaded makes the tests pass, but I didn't confirm that there are no edge cases.
* `requirement_set` is never `None` anymore? Need to confirm that, but the tests were never hitting the first branch.

File: tests/functional/conflict_test.py [hunk 2]
```diff
@@ -42,6 +42,9 @@ def test_conflicting_reqs(tmpdir):
     err = T.strip_coverage_warnings(err)
     err = T.strip_pip_warnings(err)
     assert err == (
+        "conflicting-package 1 has requirement many-versions-package<2, but you'll "
+        'have many-versions-package 3 which is incompatible.\n'
+        # TODO: do we still need to append our own error?
```
Comment (chriskuehl) [inline tests/functional/conflict_test.py]: related: #210

File: tests/functional/simple_test.py [hunk 3]
```diff
@@ -409,10 +409,8 @@ def test_override_requirements_file(tmpdir):
         'bootstrap-deps=', '-r', 'requirements-bootstrap.txt',
     )
     err = strip_pip_warnings(err)
-    assert err == (
-        'You must give at least one requirement to install '
-        '(see "pip help install")\n'
-    )
+    # pip>=10 doesn't complain about installing an empty requirements file.
+    assert err == ''
```
Comment (chriskuehl) [inline tests/functional/simple_test.py]: pretty sure this assertion (and the other similar two below) wasn't actually trying to test this specific behavior, it was just asserting the error output as a sanity check during the larger behavior test.

File: pip_faster.py (hunk 1 above)
Comment (chriskuehl) [reply pip_faster.py]: I couldn't find any examples where `extraneous` was calculated differently based on subtracting out `requirement_set.successfully_installed` even on old venv-update (pre-pip-10); all the tests pass even after I calculate `extraneous` both with and without that subtraction and then assert their equality.

I've deleted that from the code for now since I can't find any reason not to.

@bukzor do you know by any chance if/why it's necessary?
